src/Management/src/EndpointBase/Metrics/Observer/EventCounterListener.cs

```diff
@@ -70,18 +83,37 @@ protected override void OnEventSourceCreated(EventSource eventSource)
 
         if (_eventSourceName.Equals(eventSource.Name, StringComparison.OrdinalIgnoreCase))
         {
-            var refreshInterval = new Dictionary<string, string>() { { "EventCounterIntervalSec", "1" } }; // TODO: Make it configurable
-            try
+            if (!_isInitialized)
             {
-                EnableEvents(eventSource, EventLevel.Verbose, EventKeywords.All, refreshInterval);
+                _eventSources.Add(eventSource);
             }
-            catch (Exception ex)
+            else
             {
-                _logger?.LogError(ex.Message, ex);
+                SafeEnableEvents(eventSource);
             }
         }
     }
 
+    private void ProcessPreInitEventSources()
+    {
+        foreach (var eventSource in _eventSources)
+        {
+            SafeEnableEvents(eventSource);
+        }
+    }
+
+    private void SafeEnableEvents(EventSource eventSource)
+    {
+        try
+        {
+            EnableEvents(eventSource, EventLevel.Verbose, EventKeywords.All, _refreshInterval);
+        }
+        catch (Exception ex)
+        {
+            _logger?.LogError(ex.Message, ex);
```

Comment:
```suggestion
            _logger?.LogError($"Failed to enable events: {ex.Message}", ex);
```

Reply:
Thanks for the update. Next time, can you let me resolve the conversation I initiated, so I can verify the change?

Reply:
I accepted your change and I believe it automatically resolved it.

Reply:
Oh, I didn't know it worked that way. In that case, it's fine.